scripts/lint-go-mod-version.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+# SPDX-FileCopyrightText: 2023 The Pion community <https://pion.ly>
+# SPDX-License-Identifier: MIT
+
+set -e
+
+SCRIPT_PATH=$(
+  cd "$(dirname "${BASH_SOURCE[0]}")"
+  pwd -P
+)
+
+if [ -f ${SCRIPT_PATH}/.ci.conf ]; then
+  . ${SCRIPT_PATH}/.ci.conf
+fi
+
+GO_MOD_VERSION_EXPECTED=${GO_MOD_VERSION_EXPECTED:-1.18}
```

Comment:
Do we need to update the expected version manually?

Reply:
Currently yes. I was considering using some organisation wide CI variables for configures this. What do you think?

Reply:
I think the default value is better to be synchronized with
https://github.com/pion/.goassets/blob/98038b53eb3b5ada6d675022898d451255f55de5/ci/.github/workflows/test.yaml#L26
which is automatically updated in https://github.com/pion/.goassets/blob/master/.github/workflows/update-go-version.yml triggered by renovate

Reply:
Done. I've extended the `update-go-version.yml` workflow to also update shell scripts.